src/map/utils/petutils.cpp

```diff
@@ -1096,8 +1096,10 @@ namespace petutils
 
         Pet_t* PPetData = g_PPetList.at(PetID);
 
-        if (PMaster->objtype == TYPE_PC)
+        if (PMaster->objtype == TYPE_PC){
```

Comment:
Will make that change, but it turns out that it works for a single logout/login then something else goes and resets the pet info to zero. Might as well cancel this request until I figure that out >.<.